### PR TITLE
Migration to v1

### PR DESCRIPTION
**Pull request by magarwal-reef**

### CODE REVIEW

File: b2/console_tool.py [hunk 1]
```diff
@@ -898,7 +899,8 @@ class ListFileVersions(Command):
 
     def run(self, args):
         bucket = self.api.get_bucket_by_name(args.bucketName)
-        response = bucket.list_file_versions(
+        response = self.api.session.list_file_versions(
```
Comment (ppolewicz) [inline b2/console_tool.py]: I think we should make a CliBucket that inherits from bucket and adds those methods there, copy the methods content from sdk.v0

Comment (magarwal-reef) [reply b2/console_tool.py]: Done

**Comment by ppolewicz** [comment]
So... what is the difference between from v0 sync_folders and v1 sync_folders?

**Comment by magarwal-reef** [comment]
> So... what is the difference between from v0 sync_folders and v1 sync_folders?

v0 sync_folders uses the wrapped v1 synchronizer class sync_folders which is created to return v0 style exceptions for exceptions like InvalidArgument and IncompleteSync

File: b2/cli_bucket.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+from b2sdk import v1
+
+
+class CliBucket(v1.Bucket):
+    def list_file_names(self, start_filename=None, max_entries=None, prefix=None):
+        """
+        Legacy interface which just returns whatever remote API returns.
+        """
+        return self.api.session.list_file_names(self.id_, start_filename, max_entries, prefix)
```
Comment (ppolewicz) [inline b2/cli_bucket.py]: according to apiver rules, when we start using internals of b2sdk like this, we need to pin the version of b2sdk more strictly, in this case `>=1.0.0` but `<1.1.0`. Also, please add an appropriate comment in this place to indicate that
> according to b2sdk version policy, usage of B2Session requires more strict pinning.

Comment (magarwal-reef) [reply b2/cli_bucket.py]: Updated b2sdk version in requirements.txt and added appropriate doc string

**Comment by ppolewicz** [comment]
@magarwal-reef please rebase on current master

**Comment by ppolewicz** [comment]
submitting to upstream

**Pull request closed without merging**